File: AtCoder/Beginner/210/A.py
n,a,x,y=map(int,input().split())
print(n*x if n<a else a*x+(n-a)*y)

"""
Approach
If we want less number of cabbages than required discount we will buy them directly at price x
Else we will buy first "a" at price x and remaining at discounted price y
"""

"""This could have been made faster by

from sys import stdin
n,a,x,y=map(int,stdin.read().split())
print(n*x if n<a else a*x+(n-a)*y)"""

File: AtCoder/Beginner/210/A.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n,a,x,y;            // Input variables
    cin>>n>>a>>x>>y;


    // If number of cabbages are less than discount price we would directly buy them
    // Otherwise, we will buy first a at normal and then reamining at discount
    cout<<(n>=a ? a*x+(n-a)*y:n*x);
    return 0;
}
